integrated_channels/utils.py

```diff
@@ -36,7 +37,11 @@ def convert_invalid_course_ids(course_id):
     re2 = re.compile(r"[|<>.&%\s\\/\“]+")
     if re2.search(course_id):
         # If the course key contains any of the invalid chars, encode the key
-        return encode_course_key_into_base64(course_id)
+        course_id = encode_course_key_into_base64(course_id)
+    # If the encoded or unencoded version of the key are over 50 characters, they will error out
+    # in cornerstone, so we convert them to a uuid.
+    if len(course_id) > 50:
```

Comment:
Should this just be a method on the CornerstoneCourseKey class?

Reply:
For this, we would have to make different instances in the code for the other usages it has, so while the general utils file may not be the best, I don't know if its the best to only have it available as part of the model